### PR TITLE
Custom indicator developer example

### PR DESCRIPTION
**Pull request by PA-Rweins**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:  https://github.com/demisto/etc/issues/34971

## Description
This is a dummy integration to demo the CustomIndicator helper class

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorTypes/mycustomindicator.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
+{
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorTypes/mycustomindicator.json:1]: should be under `Packs/DeveloperTools/IndicatorTypes/mycustomindicator.json`

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorFields/param1.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorFields/param1.json:1]: should be under `Packs/DeveloperTools/IndicatorFields/param1.json`

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorFields/param2.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorFields/param2.json:1]: should be under `Packs/DeveloperTools/IndicatorFields/param2.json`

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/pack_metadata.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/pack_metadata.json:1]: remove

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/TestPlaybooks/CustomIndicatorDemo-test.yml [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+  "1":
+    id: "1"
+    taskid: 86b65937-519e-421c-8b04-50e0ad26d93d
+    type: regular
+    task:
+      id: 86b65937-519e-421c-8b04-50e0ad26d93d
+      version: -1
+      name: test-custom-indicator
+      description: '[Enter a description of the command, including any important information
+        users need to know, for example required permissions.]'
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/TestPlaybooks/CustomIndicatorDemo-test.yml:42]: ?

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorTypes/mycustomindicator.json [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+            "simple": "custom.param1",
+            "complex": null
+        },
+        "param2": {
+            "simple": "custom.param2",
+            "complex": null
+        }
+    },
+    "expiration": 0,
+    "layout": "cfbb8100-4832-4f58-8b7e-094c3d1b443c",
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorTypes/mycustomindicator.json:26]: upload the layout or remove this field

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+
+''' CLIENT CLASS '''
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    This Client implements API calls, and does not contain any XSOAR logic.
+    Should only do requests and return data.
+    It inherits from BaseClient defined in CommonServer Python.
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py:21]:
```suggestion
    It inherits from BaseClient defined in CommonServerPython.
```

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+    Most calls use _http_request() that handles proxy, SSL verification, etc.
+    For this  implementation, no special attributes defined
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py:23]:
```suggestion
    For this implementation, no special attributes are defined.
```

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+    try:
+        message = 'ok'
+    except DemistoException as e:
+        if 'Forbidden' in str(e) or 'Authorization' in str(e):
+            message = 'Authorization Error: make sure API Key is correctly set'
+        else:
+            raise e
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py:59]:
```suggestion
            raise
```

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+    :param Client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+    try:
+        message = 'ok'
+    except DemistoException as e:
+        if 'Forbidden' in str(e) or 'Authorization' in str(e):
+            message = 'Authorization Error: make sure API Key is correctly set'
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py:57]:
```suggestion
            message = 'Authorization Error: make sure API Key is correctly set.'
```

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+    score = Common.DBotScore.GOOD
+    indicator_value = 'custom_value'
+
+    # Create a DBotScore object
+    # Give it an indicator_type of DBotScoreType.CUSTOM
+    dbot_score = Common.DBotScore(
+        indicator=indicator_value,
+        indicator_type=DBotScoreType.CUSTOM,
+        integration_name='DummyIntegration',
+        score=score
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py:76]: [PEP8](https://www.python.org/dev/peps/pep-0008/#when-to-use-trailing-commas)
```suggestion
        score=score,
```

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+        data=data,
+        context_prefix='custom',
+    )
+    # Return a CommandResults object containing the CustomIndicator object created
+    return CommandResults(
+        readable_output='custom_value',
+        outputs=result,
+        outputs_prefix='Demo.Result',
+        outputs_key_field='test_key_field',
+        indicator=custom_indicator
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py:97]:
```suggestion
        indicator=custom_indicator,
```

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.yml [hunk 13]
```diff
@@ -0,0 +1,46 @@
...
+  required: true
+  type: 4
+- display: Trust any certificate (not secure)
+  name: insecure
+  required: false
+  type: 8
+- display: Use system proxy settings
+  name: proxy
+  required: false
+  type: 8
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.yml:23]: can be removed

**Review by yaakovi — CHANGES_REQUESTED** [review]
looks like you created a pack within a pack, this is not supported.
please update the file locations.

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorTypes/mycustomindicator.json (hunk 1 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorTypes/mycustomindicator.json:1]: Fixed

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorFields/param1.json (hunk 2 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorFields/param1.json:1]: Fixed

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorFields/param2.json (hunk 3 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorFields/param2.json:1]: Fixed

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/pack_metadata.json (hunk 4 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/pack_metadata.json:1]: fixed

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/TestPlaybooks/CustomIndicatorDemo-test.yml (hunk 5 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/TestPlaybooks/CustomIndicatorDemo-test.yml:42]: Changed to correct description

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorTypes/mycustomindicator.json (hunk 6 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/IndicatorTypes/mycustomindicator.json:26]: removed it

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py (hunk 7 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py:21]: fixed

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py (hunk 8 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py:23]: fixed

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py (hunk 9 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py:59]: fixed

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py (hunk 10 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py:57]: fixed

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py (hunk 11 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py:76]: fixed

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py (hunk 12 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py:97]: fixed

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.yml (hunk 13 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.yml:23]: removed

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.yml [hunk 14]
```diff
@@ -0,0 +1,28 @@
+category: Vulnerability Management
+commonfields:
+  id: CustomIndicatorDemo
+  version: -1
+configuration:
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.yml:5]:
```suggestion
configuration: []
```

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py [hunk 15]
```diff
@@ -0,0 +1,136 @@
...
+
+def main() -> None:
+    """main function, parses params and runs command functions
+
+    :return:
+    :rtype:
+    """
+    base_url = urljoin(demisto.params()['url'], '/api/v1')
+    verify_certificate = not demisto.params().get('insecure', False)
+    proxy = demisto.params().get('proxy', False)
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py:112]: those don't exist anymore

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/CustomIndicatorDemo_description.md [hunk 16]
```diff
@@ -0,0 +1,4 @@
+## Custom Indicator Demo Help
+
+How to configure:
+- This is a dummy integration, therefore any dummy server URL and API key will suffice.
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/CustomIndicatorDemo_description.md:4]: not relevant anymore, delete this file

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/README.md [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+2. Search for CustomIndicatorDemo.
+3. Click **Add instance** to create and configure a new integration instance.
+
+    | **Parameter** | **Description** | **Required** |
+    | --- | --- | --- |
+    | Your server URL |  | True |
+    | API Key | The API Key to use for connection | True |
+    | Trust any certificate (not secure) |  | False |
+    | Use system proxy settings |  | False |
+
```
Comment (yaakovi) [inline Packs/DeveloperTools/Integrations/CustomIndicatorDemo/README.md:16]: not relevant anymore

**Review by yaakovi — APPROVED** [review]

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/README.md (hunk 17 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/README.md:16]: fixed

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/CustomIndicatorDemo_description.md (hunk 16 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/CustomIndicatorDemo_description.md:4]: deleted

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py (hunk 15 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.py:112]: Fixed

File: Packs/DeveloperTools/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.yml (hunk 14 above)
Comment (PA-Rweins) [reply Packs/DeveloperTools/Integrations/CustomIndicatorDemo/CustomIndicatorDemo.yml:5]: fixed

**Comment by ShirleyDenkberg** [comment]
@yaakovi Doc review completed.

**Pull request merged**